sycl/doc/Array_Kernel_Parameters.md

```diff
@@ -0,0 +1,435 @@
+
+    // Call inAcc[1] accessor�s init function
+    Accessor::init(&local.inAcc[1], inAccData_1, inAccR1_1, inAccR2_1, inI_1);
+
+    callee(&local, id<1> wi);
+}
+```
+
+<h3>Fix 3: Accessor Arrays within Structs</h3>
+
+*Individual* Accessors within structs were already supported.
+Struct parameters of kernels that are structs are traversed member
+by member, recursively, to enumerate member structs that are one of
+the SYCL special types: Accessors and Samplers. For each special
+struct encountered in the scan, arguments of their init functions
+are added as separate arguments to the kernel.
+However, *arrays* of accessors within structs were not supported.
+Building on the support for single Accessors within structs,
+the extension to arrays of Accessors/Samplers within structs
+is straightforward. Each element of such arrays is treated as
+an individual object, and the arguments of its init function
+are added to the kernel arguments in sequence.
+Within the kernel caller function, the lambda object is reassembled
+in a manner similar to other instances of Accessor arrays. 
+
+
+<h4>Source code fragment:</h4>
+
+```C++
+ myQueue.submit([&](handler &cgh) {
+   using Accessor =
+        accessor<int, 1, access::mode::read, access::target::global_buffer>;
+   struct S {
+     int m;
+     Accessor inAcc[2];
+   } s = { 55,
+           {in_buffer1.get_access<access::mode::read>(cgh),
+            in_buffer2.get_access<access::mode::read>(cgh)}
+   };
+   auto outAcc = out_buffer.get_access<access::mode::write>(cgh);
+
+   cgh.parallel_for<class Worker>(num_items, [=](cl::sycl::id<1> index) {
+     outAcc[index] = s.m + s.inAcc[0][index] + s.inAcc[1][index];
+   });
+});
+```
+
+<h4>Integration header:</h4>
+
+```C++
+static constexpr
+const kernel_param_desc_t kernel_signatures[] = {
+  //--- _ZTSZZ4mainENKUlRN2cl4sycl7handlerEE20->18clES2_E6Worker
+  { kernel_param_kind_t::kind_accessor, 4062, 0 },
+  { kernel_param_kind_t::kind_std_layout, 72, 32 },
+  { kernel_param_kind_t::kind_accessor, 4062, 40 },
+  { kernel_param_kind_t::kind_accessor, 4062, 72 },
+
+};
+```
+
+<h4>Device code generated in pseudo-code form:</h4>
+
+```C++
+struct Capture {
+    Accessor outAcc;
+    struct S s;
+    () {
+        // Body
+    }
+}
+
+spir_kernel void caller(
+    __global int* outAccData,  // args of OutAcc
+    range<1> outAccR1,
+    range<1> outAccR2,
+    id<1> outI,
+    struct S s,                // the struct S
+    __global int* inAccData_0, // args of s.inAcc[0]
+    range<1> inAccR1_0,
+    range<1> inAccR2_0,
+    id<1> inI_0,
+    __global int* inAccData_1, // args of s.inAcc[1]
+    range<1> inAccR1_1,
+    range<1> inAccR2_1,
+    id<1> inI_1,
+)
+{
+    // Local capture object
+    struct Capture local;
+
+    // Reassemble capture object from parts
+    // Copy struct argument contents to local copy
+    // Accessor array will be initialized by calling init functions
```

Comment:
Sounds like all three is done by the single line below (not the intent, right?) Maybe some grouping using 1) 2)...

Reply:
OK.